fastlane/lib/fastlane/plugin_manager.rb

```diff
@@ -0,0 +1,201 @@
+      content += "\n#{self.class.code_to_attach}\n"
+
+      File.write(path_to_gemfile, content)
+    end
+
+    #####################################################
+    # @!group Installing and updating dependencies
+    #####################################################
+
+    # Warning: This will exec out
+    # This is necessary since the user might be prompted for their password
+    def install_dependencies!
+      # Using puts instead of `UI` to have the same style as the `echo`
+      puts "Installing plugin dependencies..."
+      ensure_plugins_attached!
+      with_clean_bundler_env do
+        exec("bundle install --quiet && echo 'Successfully installed plugins'")
+      end
+    end
+
+    # Warning: This will exec out
+    # This is necessary since the user might be prompted for their password
+    def update_dependencies!
+      puts "Updating plugin dependencies..."
+      ensure_plugins_attached!
+      with_clean_bundler_env do
+        exec("bundle update --quiet && echo 'Successfully updated plugins'")
+      end
+    end
+
+    def with_clean_bundler_env
+      # There is an interesting problem with using exec to call back into Bundler
+      # The `bundle ________` command that we exec, inherits all of the Bundler
+      # state we'd already built up during this run. That was causing the command
+      # to fail, telling us to install the Gem we'd just introduced, even though
+      # that is exactly what we are trying to do!
+      #
+      # Bundler.with_clean_env solves this problem by resetting Bundler state before the
+      # exec'd call gets merged into this process.
+
+      Bundler.with_clean_env do
+        yield if block_given?
+      end
+    end
+
+    #####################################################
+    # @!group Initial setup
+    #####################################################
+
+    def setup
+      UI.important("It looks like fastlane plugins are not yet set up for this project.")
+
+      path_to_gemfile = gemfile_path || DEFAULT_GEMFILE_PATH
+
+      if gemfile_content.to_s.length > 0
+        UI.important("fastlane will modify your existing Gemfile at path '#{path_to_gemfile}'")
+      else
+        UI.important("fastlane will create a new Gemfile at path '#{path_to_gemfile}'")
+      end
+
+      UI.important("This change is neccessary for fastlane plugins to work")
+
+      unless UI.confirm("Can fastlane modify the Gemfile at path '#{path_to_gemfile}' for you?")
+        UI.important("Please add the following code to '#{path_to_gemfile}':")
+        puts ""
+        puts self.class.code_to_attach.magenta # we use `puts` instead of `UI` to make it easier to copy and paste
+        UI.user_error!("Please update '#{path_to_gemfile} and run fastlane again")
+      end
+
+      attach_plugins_to_gemfile!(path_to_gemfile)
+      UI.success("Successfully modified '#{path_to_gemfile}'")
+      return true
```

Comment:
I don't think we ever care about the return value of this method, so I would suggest we take out the booleans that are describing success.